Improve error handling for WebUSB issues

microbit-robert opened the pull request:
Show dialogs with updated and more user friendly text and graphics instead of a toast notification.
Continue to use existing translations until new text has been translated.

See https://microbit-global.monday.com/boards/1125389526/views/244984/pulses/1434828425 (private)

microbit-matt-hillsdon commented on lang/ui.en.json:
```diff
@@ -1116,29 +1116,21 @@
   },
   "webusb-error-clear-connect-description-2": {
-    "defaultMessage": "Close any other tabs that may be using WebUSB (e.g. MakeCode, Python Editor), or unplug and replug the micro:bit before trying again.",
+    "defaultMessage": "Close any other tabs that may be using WebUSB (e.g. MakeCode, Python Editor, machine learning tool), or unplug and replug the micro:bit before trying again.",
```
I'm not sure that we want to add the ML tool here but one to confirm on Monday.

microbit-robert replied:
This is still unresolved, but I've simplified this back to being just MakeCode and the Python Editor for now.

microbit-robert commented on lang/ui.en.json:
```diff
@@ -1116,29 +1116,21 @@
   },
   "webusb-error-reconnect-microbit-description": {
-    "defaultMessage": "Please reconnect your micro:bit and try again.",
+    "defaultMessage": "<p>Please:</p><ul><li>check that this micro:bit does not have a battery pack connected</li><li>unplug and replug the USB cable</li></ul>",
```
Do we want to update this text so it aligns more closely with what is shown in the video?

microbit-robert replied:
I've done this, as noted in the screenshots below.

jaustin commented:
> ### Fallback error requesting reconnect
> ![image](https://private-user-images.githubusercontent.com/44397098/323985362-8d90c462-584b-448c-8a28-36e0a147d8cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ0MDMzNDAsIm5iZiI6MTcxNDQwMzA0MCwicGF0aCI6Ii80NDM5NzA5OC8zMjM5ODUzNjItOGQ5MGM0NjItNTg0Yi00NDhjLThhMjgtMzZlMGExNDdkOGNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDE1MDQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY3ZGIxMTNhZWM0ODFkNGE0ZjY0N2E3YTk3NjViMTY0Nzg3MTA0MmZiYWZhOTkyNWU3YWIzZjA0ZmYyYmI0OTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PRCd6kgSbmNpxK7wdOg2C8Op9xUbQT3pSAuv7lpIOnk)
> 
> * should this be a V2 micro:bit in the video?

Sure, I will update the video when I get a moment. Is it okay to swap the assets out at that point? How disruptive will it be if dimensions change slightly, etc?

I will also make sure '1' fades and '2' appears at the same time, as that was bothering me. Any other video feedback?

> 
> ### Failed to claim interface
> ![image](https://private-user-images.githubusercontent.com/44397098/323983913-dd3d3023-50f8-47ee-ba9d-bcd94cbc1b7c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ0MDMzNDAsIm5iZiI6MTcxNDQwMzA0MCwicGF0aCI6Ii80NDM5NzA5OC8zMjM5ODM5MTMtZGQzZDMwMjMtNTBmOC00N2VlLWJhOWQtYmNkOTRjYmMxYjdjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDE1MDQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2YjM2ZTdhNDM1MzJjMzc2MTEwNzk0NWExZmFjYzBiMjE5OWViMDA2YWFjNTBmNmJjOWIzNzMzNDI0YTNhYzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0rUOTIxfPnEXoh0iDiyEyaeQZxCIQ3aZoy9K8TeB1lQ)
> 
> Should we change the language to not use process and maybe avoid the repetition? For example:
> 
> ```
> Another app or browser tab is connected to this micro:bit
> 
> Close any other tabs that may be using WebUSB (e.g. MakeCode, Python Editor, machine learning tool), or unplug and replug the micro:bit before trying again.
> ```

That language sounds better. To be even more helpful for the user, could we include favicon hints for each? 

![image](https://github.com/microbit-foundation/python-editor-v3/assets/147113/836320bc-e9c4-41b8-8c26-2a3dc58518c7)

![image](https://github.com/microbit-foundation/python-editor-v3/assets/147113/dfae78f7-f161-453f-a4ae-e5508b136a9d)

![image](https://github.com/microbit-foundation/python-editor-v3/assets/147113/0a268168-b963-4c9d-adc9-2f0244b50e77)


> 
> Is app the best option? "application"? I think "program" isn't quite right especially in the context of a micro:bit editor and process is too techy.

I think conventionally it would be a 'page' right? Another  "Another page or browser tab is connected to this micro:bit" (I think even web apps are still web pages?)

> 
> Should this have the full replug advice + video? Leaning towards no as the best correction is likely to close the other tab.
> 

I agree no. I think it's common you need to find that other tab. Though I don't know when it's MakeCode or not. I *think* it's when you have multiple windows and MakeCode is the frontmost tab in one of them?

> ### Connection timeout
> ![image](https://private-user-images.githubusercontent.com/44397098/323984795-2cb08641-3ebf-48fa-9b55-881e161c2d69.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ0MDMzNDAsIm5iZiI6MTcxNDQwMzA0MCwicGF0aCI6Ii80NDM5NzA5OC8zMjM5ODQ3OTUtMmNiMDg2NDEtM2ViZi00OGZhLTliNTUtODgxZTE2MWMyZDY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDE1MDQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwNzU1NzEwY2E4ZDIxMTkwZjdmZmJmMzBmZTBhMzU3YWM3YWQ2Zjc3ZTQxY2UwNWU1YTY2YjE4MjljMWU4ZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bCYE6CPbYsjTn0k3KW_mHvh5sSf3DzDKPHiWIAtcQgc)
> 
> I think this should have the replug advice + video.

Agree

microbit-robert commented:
Updated screenshots:

I have updated the text for this one so matches what the video shows. It's now quite tall and will likely scroll vertically on screen heights of 768px.
![image](https://github.com/microbit-foundation/python-editor-v3/assets/95928279/6b7aea4f-b00b-4078-8466-5093ed9483de)

Improved title, simplified the description to lose reference to the machine learning tool.
![image](https://github.com/microbit-foundation/python-editor-v3/assets/95928279/58cac69f-4b3e-4401-9b20-4f352c073c13)

Timeout error is now identifical to the reconnect error with a different title. Again, it's tall.
![image](https://github.com/microbit-foundation/python-editor-v3/assets/95928279/34d52a99-a736-4b63-a759-23e5d02d45a5)

microbit-robert commented:
Alternative layouts

Horizontal:
![image](https://github.com/microbit-foundation/python-editor-v3/assets/95928279/3ca696fa-34a6-429a-9e8c-e12ac4f182cd)

Vertical with a lot of horizontal padding around the video:
![image](https://github.com/microbit-foundation/python-editor-v3/assets/95928279/ee921e93-a022-4721-afa3-8c8f1b288f3a)
